packages/office-addin-cli/src/commands.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { logErrorMessage } from "./log";
+import { usageDataObject } from "./defaults";
+
+export async function upgrade(manifestPath: string) {
+  try {
+    // Convert manifest .xml to manifest.json
```

Comment:
I would suggest having this functionality be in office-addin-manifest, e.g. in a function convertManifestToJson().

The CLI command would normally part of that package. There was an ideal that office-addin-cli could expose a CLI across the packages to make it easier but that wasn't ever done, and the package service to just have some CLI utility code which can be shared across the other packages.

Reply:
"upgrade" is a very generial term, so it might be better to use something more specific. I could see the user typing: `npx office-addin-manifest convert manifest.xml`

Reply:
Usually with NPM packages, it is related to upgrading the package version, so I worry about user confusion if upgrade is used as the term for converting the manifest from an xml format to a json format.

Reply:
The goal of this command is more than just converting the manifest to json.  I've already created a command (that needs to get expanded) in office-addin-manifest that will do the conversion part, but the idea for MetaOS is that the projects will go through a onetime conversion that includes converting the manifest to json and updating package.json to have the right supporting packages and possibly some other things (if it's discovered that more is needed).  Since it's targeting the project it doesn't make sense to put it in the package targeting the manifest.

We could see that the cli package ended up being a place for a few utility functions and that it was going to be the central cli.  While the usage data package is currently targeting just telemetry we thought it made sense for it to be a general logging package and include logging to the console, or telemetry, or other places if need be and this would allow us to start making the cli package a central project cli and possibly have it be the point of entry rather than the various package entry points we have today (not sure if or how far we'd take that idea).

Reply:
The verb 'upgrade' could potentially be confusing . . . any thoughts on a better term?

Reply:
> The verb 'upgrade' could potentially be confusing . . . any thoughts on a better term?

@akrantz said `convert` previously. I think it is already a better term, but maybe `convertProject` would be more meaningful.

Reply:
I think the cli should give context so that detailed verbs aren't needed.  We don't say "office-addin-mainfest validateManifest manifest.xml" we say "office-addin-manifest validate manifest.xml" because the context of manifest is understood.  If we're going to use this as a central cli then it would be nice if it could be understood that the context is the Office Addin project.

Reply:
Resolving this thread as the name is now changed to `convert`